datafusion/core/src/physical_plan/filter.rs

```diff
@@ -231,6 +245,39 @@ impl RecordBatchStream for FilterExecStream {
     }
 }
 
+/// Return the equals Column-Pairs and Non-equals Column-Pairs
+fn collect_columns_from_predicate(predicate: &Arc<dyn PhysicalExpr>) -> EqualAndNonEqual {
```

Comment:
I think seeing  examples of this code working in tests would help verify its correctness